### PR TITLE
tests/typing: Add tests for conversions to/from dictionary

### PR DESCRIPTION
**Pull request by nbraud**
Second take on #104 ; still based on #101.

### CODE REVIEW

File: ppb_vector/vector2.py [hunk 1]
```diff
@@ -94,6 +107,9 @@ def length(self) -> float:
         # benefit, according to microbenchmarks.
         return hypot(self.x, self.y)
 
+    def asdict(self) -> typing.Mapping[str, float]:
```
Comment (nbraud) [inline ppb_vector/vector2.py]: Should this be a `property`?

**Comment by nbraud** [comment]
Rebased on #101, removed a bit of fluff from the dictionary test.

File: ppb_vector/vector2.py (hunk 1 above)
Comment (astronouth7303) [reply ppb_vector/vector2.py]: Not normally.

**Comment by astronouth7303** [comment]
Can you rebase again? Looks like another branch is still mixed in.

File: ppb_vector/vector2.py (hunk 1 above)
Comment (nbraud) [reply ppb_vector/vector2.py]: OK.

**Comment by nbraud** [comment]
@astronouth7303 That's because Github doesn't notice you merged most of those commits as part of #101.  I can rebase indeed.

**Comment by nbraud** [comment]
Done.

**Pull request merged**